dvc/repo/experiments/__init__.py

```diff
@@ -192,23 +191,30 @@ def _stash_exp(
                         self._update_params(params)
 
                     if resume_rev:
+                        if branch:
+                            branch_name = ExpRefInfo.from_ref(branch).name
+                        else:
+                            branch_name = ""
                         if self.scm.is_dirty():
-                            logger.debug(
-                                "Resume new checkpoint branch with "
-                                "modifications"
+                            logger.info(
+                                "Modified checkpoint experiment based on "
```

Comment:
What about following scenario:
1. Someone has `checkpoint` experiment
2. It runs for some epochs in workspace
3. We edit workspace (eg code) and run `dvc exp run`
4. `dvc exp apply`
5. We have repo state which we will not be able to reproduce.
Do we intend to allow that? Maybe we should warn user here that it might result in unreproducible experiment?

Reply:
Editing code in the workspace is supported, and this essentially creates a new branch from the last applied/run checkpoint commit. So if you run 5 epochs, and then want to make a branch from the 3rd epoch w/some modified params or code, the workflow is to do 

```
dvc exp run  # initial run to generate 5 epochs
dvc exp apply <SHA_for_3rd_epoch_commit>
vim script.py  # edit code
dvc exp run # run some more epochs w/that modification
```

at the end of this, you'd end up with 2 separate checkpoint refs (git branches) that share the same first 3 epochs (commits) and diverge after that point

This is all still reproducible (as long as the user's code for a single epoch is reproducible) and internally is essentially the same thing as doing `dvc repro` + `git commit` 5 times (once for each epoch), then checking out the 3rd commit, and starting a new git branch